engine/src/main/java/com/datatorrent/stram/client/StramAgent.java

```diff
@@ -421,7 +421,11 @@ private StramWebServicesInfo retrieveWebServicesInfo(String appId)
         permissionsInfo = new JSONObject(IOUtils.toString(is));
       } catch (IOException ex) {
         // ignore if file is not found
-        LOG.info("Exception in accessing permissions.json", ex);
+        if (LOG.isDebugEnabled()) {
```

Comment:
@PramodSSImmaneni +1. IMO, before the feature can be widely used and we can treat it in a good faith and assume that the file should exist, the functionality needs to be hardened. There should be at minimum check for security configuration before code tries to access permissions.json as otherwise, it has no effect. When security is disabled and the file exists, there should be a warning as permissions are not limited and it is not what devops may expect.